src/handle-checkout-mutation.js

```diff
@@ -1,15 +1,19 @@
 export default function handleCheckoutMutation(mutationRootKey, client) {
-  return function({data, model}) {
+  return function({data, errors, model}) {
     const rootData = data[mutationRootKey];
     const rootModel = model[mutationRootKey];
-    const userErrors = rootData.userErrors;
 
-    if (userErrors.length) {
-      return Promise.reject(new Error(JSON.stringify(userErrors)));
+    if (!rootData) {
+      if (errors && errors.length) {
+        return Promise.reject(new Error(JSON.stringify(errors)));
+      }
+    } else if (rootData.userErrors && rootData.userErrors.length) {
+      return Promise.reject(new Error(JSON.stringify(rootData.userErrors)));
```

Comment:
This may be more robust as

```javascript
if (rootData && rootData.checkout) {
  return client.fetchAllPages(rootModel.checkout.lineItems, {pageSize: 250}).then((lineItems) => {
    rootModel.checkout.attrs.lineItems = lineItems;
    rootModel.checkout.errors = errors;
    rootModel.checkout.userErrors = rootModel.userErrors;

    return rootModel.checkout;
  });
}

if (errors && errors.length) {
  return Promise.reject(new Error(JSON.stringify(errors)));
}

if (rootData && rootData.userErrors && rootData.userErrors.length) {
  return Promise.reject(new Error(JSON.stringify(rootData.userErrors)));
}

return Promise.reject(new Error(`The ${mutationRootKey} mutation failed due to an unknown error.`));
```